### ISSUE TITLE
Ultrasonic cannot connect, problems with getLicense call when using Java 11

### ISSUE THREAD

**Issue by masterblaster79** [airsonic/airsonic#1214]
## Problem description

The Ultrasonic client always bails out with "A network error occurred" although the network connectivity is just fine.
The problem seems to be the "licenseExpires" date string returned by the getLicense call. Ultrasonic throws an InvalidFormatException:
```
com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `java.util.Calendar` from String "292278994-08-17T07:12:55.807Z": not a valid representation (error: Failed to parse Date value '292278994-08-17T07:12:55.807Z': Cannot parse date "292278994-08-17T07:12:55.807Z": not compatible with any of standard forms ("yyyy-MM-dd'T'HH:mm:ss.SSSZ", "yyyy-MM-dd'T'HH:mm:ss.SSS", "EEE, dd MMM yyyy HH:mm:ss zzz", "yyyy-MM-dd"))
```

TCPDUMP (IP / Hostname anonymized):
```
GET /airsonic/rest/getLicense.view?u=username&c=Ultrasonic&f=json&v=1.15.0&t=ea24c404606bb077991786edc81a36c9&s=jpx4eg676o4eego67j71pnpnhn HTTP/1.1
Host: example.com
X-Real-IP: 1.2.3.4
X-Forwarded-For: 5.6.7.8, 9.10.11.12
X-Forwarded-Host: example.com
X-Forwarded-Proto: https
accept-encoding: gzip
user-agent: okhttp/3.10.0

HTTP/1.1 200 
Set-Cookie: JSESSIONID=6AC0FA14725AC7BE50E17A599C02BAB8; Path=/airsonic; HttpOnly
Access-Control-Allow-Origin: *
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: 0
X-Frame-Options: SAMEORIGIN
Content-Type: application/json;charset=UTF-8
Transfer-Encoding: chunked
Date: Mon, 26 Aug 2019 07:36:43 GMT

12c
{
   "subsonic-response" : {
      "status" : "ok",
      "version" : "1.15.0",
      "license" : {
         "valid" : true,
         "email" : "airsonic@github.com",
         "licenseExpires" : "292278994-08-17T07:12:55.807Z",
         "trialExpires" : "292278994-08-17T07:12:55.807Z"
      }
   }
}
0
```

adb logcat snippet:
```
08-26 09:50:00.680 14652 18803 I KOIN    : Resolve class[org.moire.ultrasonic.service.MusicService] with Bean[name='OnlineMusicService', class=org.moire.ultrasonic.service.MusicService, scope:CurrentMusicService]
08-26 09:50:01.763 14652 14652 W BackgroundTask: Got exception: com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `java.util.Calendar` from String "292278994-08-17T07:12:55.807Z": not a valid representation (error: Failed to parse Date value '292278994-08-17T07:12:55.807Z': Cannot parse date "292278994-08-17T07:12:55.807Z": not compatible with any of standard forms ("yyyy-MM-dd'T'HH:mm:ss.SSSZ", "yyyy-MM-dd'T'HH:mm:ss.SSS", "EEE, dd MMM yyyy HH:mm:ss zzz", "yyyy-MM-dd"))
08-26 09:50:01.763 14652 14652 W BackgroundTask:  at [Source: (okhttp3.ResponseBody$BomAwareReader); line: 8, column: 29] (through reference chain: org.moire.ultrasonic.api.subsonic.response.LicenseResponse["license"]->org.moire.ultrasonic.api.subsonic.models.License["licenseExpires"])
08-26 09:50:01.763 14652 14652 W BackgroundTask: com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `java.util.Calendar` from String "292278994-08-17T07:12:55.807Z": not a valid representation (error: Failed to parse Date value '292278994-08-17T07:12:55.807Z': Cannot parse date "292278994-08-17T07:12:55.807Z": not compatible with any of standard forms ("yyyy-MM-dd'T'HH:mm:ss.SSSZ", "yyyy-MM-dd'T'HH:mm:ss.SSS", "EEE, dd MMM yyyy HH:mm:ss zzz", "yyyy-MM-dd"))
08-26 09:50:01.763 14652 14652 W BackgroundTask:  at [Source: (okhttp3.ResponseBody$BomAwareReader); line: 8, column: 29] (through reference chain: org.moire.ultrasonic.api.subsonic.response.LicenseResponse["license"]->org.moire.ultrasonic.api.subsonic.models.License["licenseExpires"])
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:1548)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.DeserializationContext.handleWeirdStringValue(DeserializationContext.java:910)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseDate(StdDeserializer.java:524)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseDate(StdDeserializer.java:467)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.std.DateDeserializers$DateBasedDeserializer._parseDate(DateDeserializers.java:195)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.std.DateDeserializers$CalendarDeserializer.deserialize(DateDeserializers.java:240)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.std.DateDeserializers$CalendarDeserializer.deserialize(DateDeserializers.java:205)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:529)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:528)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:417)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1280)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:326)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:529)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:528)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:417)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1280)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:326)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.ObjectReader._unwrapAndDeserialize(ObjectReader.java:1703)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:1608)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1203)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at retrofit2.converter.jackson.JacksonResponseBodyConverter.convert(JacksonResponseBodyConverter.java:32)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at retrofit2.converter.jackson.JacksonResponseBodyConverter.convert(JacksonResponseBodyConverter.java:23)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at retrofit2.ServiceMethod.toResponse(ServiceMethod.java:122)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at retrofit2.OkHttpCall.parseResponse(OkHttpCall.java:217)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at retrofit2.OkHttpCall.execute(OkHttpCall.java:180)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall.execute(ExecutorCallAdapterFactory.java:91)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at org.moire.ultrasonic.service.RESTMusicService.isLicenseValid(RESTMusicService.java:149)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at org.moire.ultrasonic.service.CachedMusicService.isLicenseValid(CachedMusicService.java:99)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at org.moire.ultrasonic.activity.SelectAlbumActivity$LoadTask.doInBackground(SelectAlbumActivity.java:1132)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at org.moire.ultrasonic.activity.SelectAlbumActivity$LoadTask.doInBackground(SelectAlbumActivity.java:1117)
08-26 09:50:01.763 14652 14652 W BackgroundTask:        at org.moire.ultrasonic.util.TabActivityBackgroundTask$1.run(TabActivityBackgroundTask.java:37)
```

### Steps to reproduce
Try to connect with Ultrasonic android client to Airsonic server running in tomcat with below SW versions

## System information
 * **Airsonic version**: 10.5.0-SNAPSHOT – 1. August 2019
 * **Operating system**: Debian Buster with Kernel 5.2 on armhf platform
 * **Java version**: openjdk 11.0.4
 * **Proxy server**: nginx-1.14.2
 * **Client**: Ultrasonic 2.6.1 on Android 9 (LineageOS 16)

```
root@a20:~# cat /etc/debian_version
10.0
root@a20:~# uname -a
Linux a20 5.2.0-2-armmp-lpae #1 SMP Debian 5.2.9-2 (2019-08-21) armv7l GNU/Linux
root@a20:~# dpkg -l | grep -Ei jdk\|tomcat
ii  libtcnative-1:armhf                   1.2.21-1                     armhf        Tomcat native library using the Apache Portable Runtime
ii  libtomcat9-java                       9.0.16-4                     all          Apache Tomcat 9 - Servlet and JSP engine -- core libraries
ii  openjdk-11-jdk:armhf                  11.0.4+11-1~deb10u1          armhf        OpenJDK Development Kit (JDK)
ii  openjdk-11-jdk-headless:armhf         11.0.4+11-1~deb10u1          armhf        OpenJDK Development Kit (JDK) (headless)
ii  openjdk-11-jre:armhf                  11.0.4+11-1~deb10u1          armhf        OpenJDK Java runtime, using Hotspot JIT
ii  openjdk-11-jre-headless:armhf         11.0.4+11-1~deb10u1          armhf        OpenJDK Java runtime, using Hotspot JIT (headless)
ii  tomcat9                               9.0.16-4                     all          Apache Tomcat 9 - Servlet and JSP engine
ii  tomcat9-common                        9.0.16-4                     all          Apache Tomcat 9 - Servlet and JSP engine -- common files
```

## Additional notes
In the past this was working just fine.
I believe the issue was introduced with Debian update from Stretch to Buster along with Update of OpenJDK (8->11) and Tomcat (8->9).

**Comment by masterblaster79** [comment]
I downgraded to Java 8 and this works just fine. So the issue is related to Java 11

```
{
   "subsonic-response" : {
      "status" : "ok",
      "version" : "1.15.0",
      "license" : {
         "valid" : true,
         "email" : "airsonic@github.com",
         "licenseExpires" : "8994-08-17T07:12:55.807Z",
         "trialExpires" : "8994-08-17T07:12:55.807Z"
      }
   }
}
```

**Comment by ibmibmibm** [comment]
fixed in #1216

**Comment by jvoisin** [comment]
There is a workaround documented [here]( https://dustri.org/b/making-ultrasonic-work-with-airsonic-on-java-11.html), until Airsonic's next version with the fix is out.

**Comment by mjanssens** [comment]
Thanks for pointing to the workaround, I was running it for a while. But noticed that 10.4.2 release was out, expecting the fix. However the issue with Ultrasonic remains, so reverted to the workaround again for now.

**Comment by mjanssens** [comment]
Appears to be fixed now with 10.5.0, thanks
